packages/nx/src/utils/create-package-json.ts

```diff
@@ -104,64 +143,70 @@ export function createPackageJson(
 }
 
 function findAllNpmDeps(
-  projectName: string,
+  projectNode: ProjectGraphProjectNode,
   graph: ProjectGraph,
-  list: {
-    dependencies: Record<string, string>;
-    peerDependencies: Record<string, string>;
-    peerDependenciesMeta: Record<string, { optional: boolean }>;
-  } = { dependencies: {}, peerDependencies: {}, peerDependenciesMeta: {} },
-  seen = new Set<string>()
-) {
-  const node = graph.externalNodes[projectName];
-
-  if (seen.has(projectName)) {
-    // if it's in peerDependencies, move it to regular dependencies
-    // since this is a direct dependency of the project
-    if (node && list.peerDependencies[node.data.packageName]) {
-      list.dependencies[node.data.packageName] = node.data.version;
-      delete list.peerDependencies[node.data.packageName];
-    }
-    return list;
-  }
+  npmDeps: NpmDeps,
+  seen: Set<string>,
+  dependencyPatterns: string[],
+  rootPatterns?: string[]
+): void {
+  const projectFiles = filterUsingGlobPatterns(
```

Comment:
We should probably have `seen.add(dep)` + `check` on this level as well to avoid the globbing and file filtering for already checked projects.

If we have something like:
```
      A
    /   \
   B     C
    \   /
      D
```

Where A, B, C, and D and internal projects, both B and C would trigger checking of D's dependencies and double work.

Let's avoid that by adding the internal projects to seen list as well.